src/ILCompiler.Compiler/src/CppCodeGen/NodeIteration/DependencyNodeIterator.cs

```diff
@@ -0,0 +1,105 @@
+﻿using ILCompiler.DependencyAnalysis;
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.TypeSystem;
+using System.Collections.Generic;
+
+namespace ILCompiler.Compiler.CppCodeGen
+{
+    public class DependencyNodeIterator
```

Comment:
Can you add some comments about what this does? I'm not sure I follow why we need this kind of filtering of the `IEnumerable<DependencyNode> nodes` that are passed to the constructor. The enumeration of all necessary nodes coming from the dependency graph should contain everything that's needed (and nothing more/less than that).